src/main/java/com/faforever/client/chat/UserFilterController.java

```diff
@@ -124,23 +134,42 @@ boolean isGameStatusMatch(ChatUserItemController chatUserItemController) {
   }
 
   public void onGameStatusPlaying(ActionEvent actionEvent) {
-    playerStatusFilter = PLAYING;
-    gameStatusMenu.setText(i18n.get("game.gameStatus.playing"));
-    filterUsers();
+    setOrUnsetGameStatusFilter(PLAYING);
   }
 
   public void onGameStatusLobby(ActionEvent actionEvent) {
-    playerStatusFilter = LOBBYING;
-    gameStatusMenu.setText(i18n.get("game.gameStatus.lobby"));
-    filterUsers();
+    setOrUnsetGameStatusFilter(LOBBYING);
   }
 
   public void onGameStatusNone(ActionEvent actionEvent) {
-    playerStatusFilter = IDLE;
-    gameStatusMenu.setText(i18n.get("game.gameStatus.none"));
+    setOrUnsetGameStatusFilter(IDLE);
+  }
+
+  private void setOrUnsetGameStatusFilter(PlayerStatus newStatus) {
+    playerStatusFilter = newStatus == playerStatusFilter ? null : newStatus;
+    setGameStatusMenuText(playerStatusFilter);
     filterUsers();
   }
 
+  private void setGameStatusMenuText(PlayerStatus status) {
+    if (status == null) {
+      gameStatusMenu.setText(i18n.get("game.gameStatus"));
+      gameStatusToggleGroup.selectToggle(null);
```

Comment:
If you `return;` at the end of this `if`, you don't need an `else` and can avoid nested blocks and indentation.
Also, a `default` case should always be added to make sure that there is no uncovered enum value - or at least to let us know there is (if someone adds a value, he might not be aware that he needs to adjust this code accordingly, so let's tell him).

```
private void setGameStatusMenuText(PlayerStatus status) {
  if (status == null) {
    gameStatusMenu.setText(i18n.get("game.gameStatus"));
    gameStatusToggleGroup.selectToggle(null);
    return;
  }

  switch (status) {
    case PLAYING:
      gameStatusMenu.setText(i18n.get("game.gameStatus.playing"));
      break;
    case LOBBYING:
       gameStatusMenu.setText(i18n.get("game.gameStatus.lobby"));
      break;
    case IDLE:
      gameStatusMenu.setText(i18n.get("game.gameStatus.none"));
      break;
    default:
      throw new ProgrammingError("Uncovered player status: " + status);
  }
}
```

Oh and since `setGameStatusMenuText` doesn't actually **set** the value given to it but does some calculation based on the given value to eventually **update** the property with some value, I'd rather call it `updateGameStatusMenuText`.